help_function/src/util_group_store_test.cpp

```diff
@@ -0,0 +1,270 @@
+// ====------ util_group_store_test.cpp------------ *- C++ -* ----===//
+
+
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+// ===----------------------------------------------------------------------===//
+#include <dpct/dpct.hpp>
+#include <dpct/dpl_utils.hpp>
+#include <iostream>
+#include <oneapi/dpl/iterator>
+#include <sycl/sycl.hpp>
+
+
+template <dpct::group::store_algorithm S>
+bool helper_validation_function(const int *ptr, const char *func_name) {
+  for (int i = 0; i < 512; ++i) {
+    if (ptr[i] != i) {
+      if constexpr (S == dpct::group::store_algorithm::BLOCK_STORE_DIRECT) {
+      std::cout << func_name << "_blocked"
+                << " failed\n";}
+      else{
+        std::cout << func_name << "_striped"
+                << " failed\n";
+      }
+      std::ostream_iterator<int> Iter(std::cout, ", ");
+      std::copy(ptr, ptr + 512, Iter);
+      std::cout << std::endl;
+      return false;
+    }
+  }
+    if constexpr (S == dpct::group::store_algorithm::BLOCK_STORE_DIRECT) {
+      std::cout << func_name << "_blocked"
+                << " passed\n";}
+      else{
+        std::cout << func_name << "_striped"
+                << " passed\n";
+      } 
+    return true;
+}
+
+bool subgroup_helper_validation_function(const int *ptr, const uint32_t *sg_sz,
+                                         const char *func_name) {
+  
+  int expected[512];
+  int num_threads = 128;
+  int items_per_thread = 4;
+  uint32_t sg_sz_val = *sg_sz;
+  for (int i = 0; i < num_threads; ++i) {
+    for (int j = 0; j < items_per_thread; ++j) {
+      expected[items_per_thread * i + j] =
+          (i / sg_sz_val) * sg_sz_val * items_per_thread + sg_sz_val * j +
+          i % sg_sz_val;
+    }
+  }

```

Comment:
It looks like we are setting up `expected` and then never using it.  Am I missing something?